server/game/cards/03_TWI/leaders/GeneralGrievousGeneralOfTheDroidArmies.ts

```diff
@@ -0,0 +1,45 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { KeywordName, Trait, WildcardCardType } from '../../../core/Constants';
+
+export default class GeneralGrievousGeneralOfTheDroidArmies extends LeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '2872203891',
+            internalName: 'general-grievous#general-of-the-droid-armies',
+        };
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addActionAbility({
+            title: 'Give a Droid unit Sentinel for this phase',
+            cost: [AbilityHelper.costs.exhaustSelf()],
+            targetResolver: {
+                cardTypeFilter: WildcardCardType.Unit,
+                cardCondition: (card) => card.isUnit() && card.hasSomeTrait(Trait.Droid),
+                immediateEffect: AbilityHelper.immediateEffects.forThisPhaseCardEffect({
+                    effect: AbilityHelper.ongoingEffects.gainKeyword({ keyword: KeywordName.Sentinel })
+                })
+            }
+        });
+    }
+
+    protected override setupLeaderUnitSideAbilities() {
+        this.addOnAttackAbility({
+            title: 'Give a Droid unit +1/+0 and Sentinel for this phase',
+            optional: true,
+            immediateEffect: AbilityHelper.immediateEffects.selectCard({
```

Comment:
We should use a `targetResolver` here instead of `selectCard()`. `selectCard()` really only exists for cases where using `targetResolver` is not possible, such inside of another effect like `sequential()`.

Note that this will break your tests currently because the autoSingleTarget feature is not working right at the moment for `selectCard`, there will be at least one place you will need to an add explicit click on the target. I am planning to fix that.